### ISSUE TITLE
RibbonLoadBalancerClient reconstructs scheme-less URIs into http://my-app.example.com:443 

### ISSUE THREAD

**Issue by william-tran** [spring-cloud/spring-cloud-netflix#1126]
I want to use a `@LoadBalanced` RestTemplate to access `MY-APP` without needing to know how `MY-APP` was registered, eg, insecure only, secure only, or both. To achieve this, I want to leave out the scheme in the URI, eg 
```
restTemplate.getForObject("//MY_APP/api", Object.class);
```
I would expect this URI to resolve to the secure route if available, and if not, the insecure route. But this URI only works when the registration is insecure only.

If there is a secure route, that port is used as the ServiceInstance's port:

https://github.com/spring-cloud/spring-cloud-netflix/blob/v1.1.2.RELEASE/spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/ribbon/eureka/DomainExtractingServerList.java#L66-L67
https://github.com/Netflix/ribbon/blob/v2.2.0/ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/DiscoveryEnabledServer.java#L43-L44

But, if there is no specific ribbon client configuration for that service, the scheme defaults to http:
https://github.com/spring-cloud/spring-cloud-netflix/blob/v1.1.2.RELEASE/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonLoadBalancerClient.java#L56-L60

This is because the config object is never null, even when I don't have any explicit client configuration. And `config.get(CommonClientConfigKey.IsSecure, false)` defaults to false. 
https://github.com/spring-cloud/spring-cloud-netflix/blob/v1.1.2.RELEASE/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonLoadBalancerClient.java#L117

So, without any explicit client config, we have a scheme that defaults to http but a port that defaults to the secure port. 

This can be fixed by using the InstanceInfo to determine `IsSecure` if there is no explicit client configuration of `IsSecure`:
```
	private boolean isSecure(Server server, String serviceId) {
		IClientConfig config = this.clientFactory.getClientConfig(serviceId);
		if (config != null) {
			Boolean isSecure = config.get(CommonClientConfigKey.IsSecure);
			if (isSecure != null) {
				return isSecure;
			} 
		}
		return serverIntrospector(serviceId).isSecure(server);
	}
```

**Comment by spencergibb** [comment]
@william-tran thanks for detailed report. Your suggested change should go in `RibbonLoadBalancerClient` correct? If we get this done quickly, we can have it part of 1.1.3 which will be part of Brixton.SR2. Are you already working on a PR? If not, I'll work on it today.

**Comment by william-tran** [comment]
I'll submit the PR

**Comment by spencergibb** [comment]
Nice, make sure to do it against the 1.1.x branch and we'll merge into master.

**Comment by spencergibb** [comment]
Closed via 99beb9d04573b96a53006a0bbef8b63747b96169
